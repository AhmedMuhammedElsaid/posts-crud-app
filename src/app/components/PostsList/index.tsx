import React from "react";
import classes from "./index.module.css";
import PostItem from "../PostItem";
import { IPostsList } from "@/app/types";

const PostsList: React.FC<IPostsList> = ({ posts }) => {
  return (
    <ul className={classes.listWrapper}>
      {posts.map((post) => (
        <PostItem key={post.id} {...post} />
      ))}
    </ul>
  );
};

export default PostsList;
