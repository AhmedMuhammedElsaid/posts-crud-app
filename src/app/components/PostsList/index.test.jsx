import React from "react";
import { render, screen } from "@testing-library/react";
import PostsList from "./index";

test("renders posts list", () => {
  const mockPosts = [
    { id: 1, title: "Test Post 1", body: "This is a test post 1" },
    { id: 2, title: "Test Post 2", body: "This is a test post 2" },
  ];

  const { asFragment } = render(<PostsList posts={mockPosts} />);
  expect(asFragment()).toMatchSnapshot();
  mockPosts.forEach((post) => {
    expect(screen.getByText(post.title)).toBeInTheDocument();
    expect(screen.getByText(post.body)).toBeInTheDocument();
  });
});
