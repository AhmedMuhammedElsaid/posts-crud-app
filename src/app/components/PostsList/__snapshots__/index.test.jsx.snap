// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders posts list 1`] = `
<DocumentFragment>
  <ul
    class="listWrapper"
  >
    <li
      class="listItem"
      data-testid="test-1"
    >
      <h2
        class="title"
        data-testid="post-title"
      >
        Test Post 1
      </h2>
      <p
        class="body"
        data-testid="post-body"
      >
        This is a test post 1
      </p>
      <div
        class="icons"
        data-testid="icons-wrapper"
      >
        <a
          href="/post/1"
        >
          <svg
            fill="currentColor"
            height="1.5rem"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1.5rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
            />
          </svg>
        </a>
        <a
          href="/post/1/edit"
        >
          <svg
            fill="currentColor"
            height="1.5rem"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1.5rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Edit"
            >
              <g>
                <path
                  d="M3.548,20.938h16.9a.5.5,0,0,0,0-1H3.548a.5.5,0,0,0,0,1Z"
                />
                <path
                  d="M9.71,17.18a2.587,2.587,0,0,0,1.12-.65l9.54-9.54a1.75,1.75,0,0,0,0-2.47l-.94-.93a1.788,1.788,0,0,0-2.47,0L7.42,13.12a2.473,2.473,0,0,0-.64,1.12L6.04,17a.737.737,0,0,0,.19.72.767.767,0,0,0,.53.22Zm.41-1.36a1.468,1.468,0,0,1-.67.39l-.97.26-1-1,.26-.97a1.521,1.521,0,0,1,.39-.67l.38-.37,1.99,1.99Zm1.09-1.08L9.22,12.75l6.73-6.73,1.99,1.99Zm8.45-8.45L18.65,7.3,16.66,5.31l1.01-1.02a.748.748,0,0,1,1.06,0l.93.94A.754.754,0,0,1,19.66,6.29Z"
                />
              </g>
            </g>
          </svg>
        </a>
        <button
          class="delete"
          data-testid="delete-button"
        >
          <svg
            fill="currentColor"
            height="1.2rem"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1.2rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M268 416h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12zM432 80h-82.41l-34-56.7A48 48 0 0 0 274.41 0H173.59a48 48 0 0 0-41.16 23.3L98.41 80H16A16 16 0 0 0 0 96v16a16 16 0 0 0 16 16h16v336a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128h16a16 16 0 0 0 16-16V96a16 16 0 0 0-16-16zM171.84 50.91A6 6 0 0 1 177 48h94a6 6 0 0 1 5.15 2.91L293.61 80H154.39zM368 464H80V128h288zm-212-48h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12z"
            />
          </svg>
        </button>
      </div>
    </li>
    <li
      class="listItem"
      data-testid="test-2"
    >
      <h2
        class="title"
        data-testid="post-title"
      >
        Test Post 2
      </h2>
      <p
        class="body"
        data-testid="post-body"
      >
        This is a test post 2
      </p>
      <div
        class="icons"
        data-testid="icons-wrapper"
      >
        <a
          href="/post/2"
        >
          <svg
            fill="currentColor"
            height="1.5rem"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1.5rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
            />
          </svg>
        </a>
        <a
          href="/post/2/edit"
        >
          <svg
            fill="currentColor"
            height="1.5rem"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1.5rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Edit"
            >
              <g>
                <path
                  d="M3.548,20.938h16.9a.5.5,0,0,0,0-1H3.548a.5.5,0,0,0,0,1Z"
                />
                <path
                  d="M9.71,17.18a2.587,2.587,0,0,0,1.12-.65l9.54-9.54a1.75,1.75,0,0,0,0-2.47l-.94-.93a1.788,1.788,0,0,0-2.47,0L7.42,13.12a2.473,2.473,0,0,0-.64,1.12L6.04,17a.737.737,0,0,0,.19.72.767.767,0,0,0,.53.22Zm.41-1.36a1.468,1.468,0,0,1-.67.39l-.97.26-1-1,.26-.97a1.521,1.521,0,0,1,.39-.67l.38-.37,1.99,1.99Zm1.09-1.08L9.22,12.75l6.73-6.73,1.99,1.99Zm8.45-8.45L18.65,7.3,16.66,5.31l1.01-1.02a.748.748,0,0,1,1.06,0l.93.94A.754.754,0,0,1,19.66,6.29Z"
                />
              </g>
            </g>
          </svg>
        </a>
        <button
          class="delete"
          data-testid="delete-button"
        >
          <svg
            fill="currentColor"
            height="1.2rem"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1.2rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M268 416h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12zM432 80h-82.41l-34-56.7A48 48 0 0 0 274.41 0H173.59a48 48 0 0 0-41.16 23.3L98.41 80H16A16 16 0 0 0 0 96v16a16 16 0 0 0 16 16h16v336a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128h16a16 16 0 0 0 16-16V96a16 16 0 0 0-16-16zM171.84 50.91A6 6 0 0 1 177 48h94a6 6 0 0 1 5.15 2.91L293.61 80H154.39zM368 464H80V128h288zm-212-48h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12z"
            />
          </svg>
        </button>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
